source/runtime.cpp

```diff
@@ -861,7 +863,11 @@ namespace reshade
 	}
 	void runtime::save_config() const
 	{
-		ini_file config(_configuration_path);
+		save_config(_configuration_path);
+	}
+	void runtime::save_config(const filesystem::path &save_path) const
+	{
+		ini_file config(_configuration_path,save_path);
```

Comment:
I don't think this is necessary. `ini_file config(save_path)` should have the same effect (since the empty `save_config` overload passes in the correct default path) and it means the changes to the `ini_file` class can be removed.

Reply:
If there is no overload, some settings will always be lost and in certain cases some settings will be lost.

Always:
NoFontScaling
(and unknown keys)

In certain cases:
DX*_BUFFER_DETECTION